src/Domain/Client/Client.cs

```diff
@@ -1,42 +1,72 @@
 namespace InvoiceKit.Domain.Client;
 
+using System.Text;
+
 public sealed class Client
 {
     public ClientName Name { get; }
     
-    public ClientContactName ContactName { get; }
+    public ClientContactName? ContactName { get; private set; }
     
-    public ClientEmail Email { get; }
+    public ClientEmail? Email { get; private set; }
     
-    public ClientPhone Phone { get; }
+    public ClientPhone? Phone { get; private set; }
     
-    public ClientAddress Address { get; }
+    public ClientAddress? Address { get; private set; }
     
-    private Client(
-        ClientName name, 
-        ClientContactName contactName, 
-        ClientEmail email, 
-        ClientPhone phone, 
-        ClientAddress address)
+    private Client(ClientName name)
     {
         Name = name;
+    }
+
+    public static Client CreateNew(ClientName name)
+    {
+        return new Client(name);
+    }
+
+    public void SetContactName(ClientContactName contactName)
+    {
         ContactName = contactName;
+    }
+
+    public void SetEmail(ClientEmail email)
+    {
         Email = email;
+    }
+
+    public void SetPhone(ClientPhone phone)
+    {
         Phone = phone;
-        Address = address;
     }
 
-    public static Client CreateNew(ClientName name, 
-        ClientContactName contactName, 
-        ClientEmail email, 
-        ClientPhone phone, 
-        ClientAddress address)
+    public void SetAddress(ClientAddress address)
     {
-        return new Client(name, contactName, email, phone, address);
+        Address = address;
     }
     
     public override string ToString()
     {
-        return Name.ToString();
+        var builder = new StringBuilder();
+        builder.Append(Name);
+        if (ContactName is not null)
+        {
+            builder.Append("\n" + ContactName);
+        }
+
+        if (Email is not null)
+        {
+            builder.Append("\n" + Email);
+        }
+
+        if (Phone is not null)
+        {
+            builder.Append("\n" + Phone);
```

Comment:
AppendLine may be a better option here. If you're worried about a trailing blank line it can always be trimmed...

Another alternative is to create a list of components and use `string.Join('\n', parts)`